### PR TITLE
Stylograph [GenesisDAO]

### PR DESCRIPTION
**Pull request by deep-ink-ventures**
#### Project Abstract

Stylograph is a framework aimed at enhancing the functionality of substrate-based chains in the Polkadot ecosystem with plugin-like functionality.

By using the ink! smart-contracting language, it allows third parties to extend core functionality, providing developers the ability to concentrate on domain logic.

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [ ] **Level 2**:  Up to $30,000, 3 approvals
- [x] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [x] Payment details have been provided (bank details via email _or_ BTC, Ethereum (USDC/DAI) or Polkadot/Kusama (USDT) address in the application).
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).

### CODE REVIEW

**Comment by keeganquigley** [comment]
Thanks for the application @deep-ink-ventures a couple of initial comments:

- 70k seems very expensive for one pallet. Would you be willing to lower the rate here?
- If I understand correctly, you wish to just build a PoC for the pallet, and then plan to ask for treasury funding afterward for the rest? (Escrow, Vesting, and Delegation)?
- Can you add more detail into the deliverables table regarding the initial functionalities of the pallet?

File: applications/Stylograph.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+| -----: |------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| **0a.** | License                            | Apache 2.0       |
+| **0b.** | Documentation | We will provide both **inline documentation** of the code and a basic **tutorial** that explains how a can integrate the pallet and start working with |
+| **0c.** | Testing and Testing Guide | Core functions will be fully covered by comprehensive unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests.  |
+| 1. | Facade Builder  | Develop the code generator for smart contracts as specified above in the `ink! Stylograph Facade Builder` section.   |
+| 2. | Refrence Implementation           |  We will provide a sample implementation alongside with `pallet_contracts` on the Genesis DAO test chain to demonstrate the functionality.    |
```
Comment (keeganquigley) [inline applications/Stylograph.md:171]:
```suggestion
| 2. | Reference Implementation           |  We will provide a sample implementation alongside with `pallet_contracts` on the Genesis DAO test chain to demonstrate the functionality.    |
```

**Comment by deep-ink-ventures** [comment]
@keeganquigley: Thanks for your comments.

> 70k seems very expensive for one pallet. Would you be willing to lower the rate here?

Our goal is to keep the Genesis DAO Team working on this until the treasury proposal passes; would it therefore be an option to expand the scope? E.g. we could add this [frontend section](https://docs.google.com/document/d/1yyEBa43plj-PzlrEnK6JT5CKqyvBDUrYiR1N6984fhk/edit#heading=h.kvq44zt0vm04) for Genesis DAO, as it can be developed in parallel. It is on our current roadmap and solves (besides some general feature addons) for some annoying usages of polkadot.js.org for transfers and multisig actions.

> If I understand correctly, you wish to just build a PoC for the pallet, and then plan to ask for treasury funding afterward for the rest? (Escrow, Vesting, and Delegation)?

Initially we wanted to apply for a treasury proposal only, but got the feedback that the decisions times are very long and we should aim for a bridge financing to not have to pause the project. So yes, basically this part has been extracted from the treasury proposal.

> Can you add more detail into the deliverables table regarding the initial functionalities of the pallet?
Can you please elaborate what is missing? We have specified in detail what the functionalities are [here](https://github.com/deep-ink-ventures/Grants-Program/blob/master/applications/Stylograph.md#pallet_stylograph) - 
is there something unclear or missing?

**Comment by deep-ink-ventures** [comment]
@keeganquigley: Thanks for your comments.

> 70k seems very expensive for one pallet. Would you be willing to lower the rate here?

Our goal is to keep the Genesis DAO Team working on this until the treasury proposal passes; would it therefore be an option to expand the scope? E.g. we could add this [frontend section](https://docs.google.com/document/d/1yyEBa43plj-PzlrEnK6JT5CKqyvBDUrYiR1N6984fhk/edit#heading=h.kvq44zt0vm04) for Genesis DAO, as it can be developed in parallel. It is on our current roadmap and solves (besides some general feature addons) for some annoying usages of polkadot.js.org for transfers and multisig actions.

> If I understand correctly, you wish to just build a PoC for the pallet, and then plan to ask for treasury funding afterward for the rest? (Escrow, Vesting, and Delegation)?

Initially we wanted to apply for a treasury proposal only, but got the feedback that the decisions times are very long and we should aim for a bridge financing to not have to pause the project. So yes, basically this part has been extracted from the treasury proposal.

> Can you add more detail into the deliverables table regarding the initial functionalities of the pallet?
Can you please elaborate what is missing? We have specified in detail what the functionalities are [here](https://github.com/deep-ink-ventures/Grants-Program/blob/master/applications/Stylograph.md#pallet_stylograph) - 
is there something unclear or missing?

**Comment by deep-ink-ventures** [comment]
Thanks for the feedback, @Noc2 !

I added the frontend parts and benchmarking - in general you can, of course, assume the same quality standards we provide in our current grant!

As for the programming languages, the pallet would be (obvisously) `rust`.

We have a [backend](https://github.com/deep-ink-ventures/genesis-dao-frontend) that is written in `python`. It is parsing the incoming events and provides an easily to access API on top of the service. The [frontend](https://github.com/deep-ink-ventures/genesis-dao-service) is written in `typescript` and uses `next.js` / `react`.

**Review by Noc2 — APPROVED** [review]
Thanks for the quick reply. Feel free to integrate next.js / react, etc. also, into the deliverables. Independent of it, I'm already happy to go ahead with it and share it with the rest of the team.

**Review by SBalaguer — DISMISSED** [review]

**Comment by deep-ink-ventures** [comment]
@Noc2 thx, techstack added!

**Review by Noc2 — APPROVED** [review]
Re-approved

**Review by SBalaguer — APPROVED** [review]
Re-approved

**Review by keeganquigley — CHANGES_REQUESTED** [review]
Thanks for the changes @deep-ink-ventures one additional comment:

- I think the Total Estimated Duration should be 5 months instead of 3 months, correct? Can you update this?

**Comment by deep-ink-ventures** [comment]
Good catch, @keeganquigley  We aim to parallelize some of the work, but I assume it doesn't hurt if we're faster. I adjusted to 4-5 month.

**Review by keeganquigley — APPROVED** [review]
@deep-ink-ventures np, that way you won't have to file an amendment PR if it does end up taking longer. LGTM.

**Review by Noc2 — APPROVED** [review]

**Review by BenWhiteJam — APPROVED** [review]
Really looking forward to it.

**Review by gautamdhameja — APPROVED** [review]

**Pull request merged**